packages/dashboard/src/components/tasks/task-inspector.tsx

```diff
@@ -0,0 +1,243 @@
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogContent,
+  DialogTitle,
+  Divider,
+  Grid,
+  Typography,
+} from '@mui/material';
+import { CardContent, useTheme } from '@mui/material';
+import { TaskEventLog, TaskState } from 'api-client';
+import React from 'react';
+import { of, switchMap } from 'rxjs';
+import { AppControllerContext } from '../app-contexts';
+import { AppEvents } from '../app-events';
+import { RmfAppContext } from '../rmf-app';
+import { TaskInfo } from 'react-components';
+import { UserProfileContext } from 'rmf-auth';
+import { Enforcer } from '../permissions';
+import { TaskLogs } from './task-logs';
+
+export interface TableDataGridState {
+  task: TaskState | null;
+  open: boolean;
```

Comment:
Similar to the comment in `task-summary`, we can probably do away with this prop, by setting a state that is default true,
```
const [isOpen, setIsOpen] = React.useState(true);
```
and gets set to false in the `onClose` dialog prop

Reply:
Same answer as above in task summary